velox/dwio/parquet/tests/writer/ParquetWriterTest.cpp

```diff
@@ -83,6 +83,66 @@ class ParquetWriterTest : public ParquetTestBase {
   inline static const std::string kHiveConnectorId = "test-hive";
 };
 
+class ArrowMemoryPool final : public ::arrow::MemoryPool {
+ public:
+  explicit ArrowMemoryPool() : allocated_(0) {}
+
+  ~ArrowMemoryPool() = default;
+
+  ::arrow::Status Allocate(int64_t size, int64_t alignment, uint8_t** out)
+      override {
+    *out = reinterpret_cast<uint8_t*>(malloc(size));
```

Comment:
Thanks for the update. Could you elaborate on how we can know whether the memory allocation is being tracked?

Reply:
@rui-mo Yes. Added check to ensure the out is not nullptr.